### ISSUE TITLE
Quoted boolean (string) value converted to boolean type when referenced in helm chart template expression and value supplied by Rancher CLI `--values` parameter

### ISSUE THREAD

**Issue by axeal** [rancher/rancher#21083]
**What kind of request is this (question/bug/enhancement/feature request):**

Bug

**Steps to reproduce (least amount of steps as possible):**

1. Create a simple Rancher chart in a custom repository, with a templated deployment containing the following test to set an environment variable when the value test.myBooleanValue.bool1 equals "true":

```
{{- if (eq .Values.test.myBooleanValue.bool1 "true") }}
  - name: BOOLEAN_TEST
    value: "IT_WORKED"
{{- end }}
```

2. Deploy the application via the Rancher CLI `rancher app install nginx-test nginx-test-quoted-boolean-test-cli --values values.yaml` where the contents of values.yaml is as follows:

```
test:
  myBooleanValue:
    bool1: "true"
```

**Result:**

Inspect the application via the Rancher API and observe the following error message:

```
`wait helm template failed. Error: render error in \"nginx-test/templates/deployment.yaml\": template: nginx-test/templates/deployment.yaml:41:18: executing \"nginx-test/templates/deployment.yaml\" at <eq .Values.test.myBo...>: error calling eq: incompatible types for comparison\n: exit status 1"`
```

**Other details that may be helpful:**

Deploy the same chart with the same quoted boolean, via the Rancher UI, providing the values via the 'Edit as Yaml' section under Answers, and the deployment is successful.

Testing rendering of the template via the helm CLI directly with the same values provided with the `--values` parameter, the template renders successfully.

**Environment information**
- Rancher v2.2.4
- Rancher CLI v2.2.0

**Comment by cjellick** [comment]
firs step will be to compare the post that the cli sends to the post that the ui sends

**Comment by deniseschannon** [comment]
@brendarearden Please feel free to reach out to @leodotcloud who had done some initial research on this bug. Findings from him ```
time="2019-09-10T19:59:20-07:00" level=info msg="{\"test\":{\"myBooleanValue\":{\"bool1\":true}}}"
time="2019-09-10T19:59:20-07:00" level=info msg="holder: map[test:map[myBooleanValue:map[bool1:true]]]"
time="2019-09-10T19:59:20-07:00" level=info msg="boolAnswers: map[test:map[myBooleanValue:map[bool1:true]]]"
time="2019-09-10T19:59:20-07:00" level=info msg="{\"test\":{\"myBooleanValue\":{\"bool1\":\"true\"}}}"
time="2019-09-10T19:59:20-07:00" level=info msg="holder: map[test:map[myBooleanValue:map[bool1:true]]]"
time="2019-09-10T19:59:20-07:00" level=info msg="strAnswers: map[test:map[myBooleanValue:map[bool1:true]]]"

The internal logic converts the yaml to json and during this phase, we are losing the string to boolean.
```

**Comment by leodotcloud** [comment]
Related: https://github.com/rancher/rancher/issues/20704

**Comment by mrajashree** [comment]
I had tested this with https://github.com/rancher/cli/pull/254 but that commit didn't fix this bug. 
When I was debugging I was considering not parsing the values.yaml file into app.Answers, instead reading it and adding it to app.ValuesYaml. Because the same chart works through rancher ui, and the ui adds the yaml to this ValuesYaml field on app.
So maybe something like
```
+         if ctx.String("values") != "" {
+                 bytes, err := ioutil.ReadFile(ctx.String("values"))
+                 if err != nil {
+                         return err
+                 }
+                 app.ValuesYaml = string(bytes)
+         }
```
over https://github.com/rancher/cli/blob/master/cmd/app.go#L686

instead of calling parseValuesFile in processAnswers

**Comment by brendarearden** [comment]
available for testing in v2.4.3-rc1

**Comment by rancher-max** [comment]
#### Validated using rancher/cli v2.4.3-rc1

- Created https://github.com/rancher-max/helm-foo for testing.
- Recreated the issues using v2.4.0 cli
- Validated it is fixed using v2.4.3-rc1
- Also working in the UI.

This was done on v2.4-2745-head rancher server.
